### PR TITLE
Project ARES: Announcement Signatures

### PR DESCRIPTION
**Pull request by realforest2001**
# About the pull request
ARES now logs who sent an announcement and not just the contents of it.

# Explain why it's good for the game

Makes the available information a little more useful. It's all well and good recording what was said, but a little pointless if not including who said it.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: Added the name of an announcement sender, if one exists, to ARES logs.
/:cl:

### CODE REVIEW

File: code/defines/procs/announcement.dm [hunk 1]
```diff
@@ -47,9 +47,9 @@
 
 		switch(logging)
 			if(ARES_LOG_MAIN)
-				log_ares_announcement(title, message)
+				log_ares_announcement(title, message, signature)
 			if(ARES_LOG_SECURITY)
-				log_ares_security(title, message)
+				log_ares_security(title, message, signature)
```
Comment (Drulikar) [inline code/defines/procs/announcement.dm:52]: This should be replicated for `/proc/shipwide_ai_announcement(message, title = MAIN_AI_SYSTEM, sound_to_play = sound('sound/misc/interference.ogg'), signature, ares_logging = ARES_LOG_MAIN)` as well 
image

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/defines/procs/announcement.dm (hunk 1 above)
Comment (Drulikar) [reply code/defines/procs/announcement.dm:52]: You also may consider updating all other usages of `log_ares_security` wherever applicable you have a user.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by realforest2001** [comment]
Added the bits for security log, had to update some var names on other UIs to do so mind.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/game/machinery/ARES/ARES_interface.dm
Comment (Drulikar) [inline code/game/machinery/ARES/ARES_interface.dm:1]: In this file there is still:
`log_ares_security("Nerve Gas Release", "[last_login] released Nerve Gas from Vent '[sec_vent.vent_tag]'.")`

File: code/defines/procs/announcement.dm
Comment (Drulikar) [inline code/defines/procs/announcement.dm:1]: In this file there is still:
`log_ares_security("[MAIN_AI_SYSTEM] Security Update", message)`
`log_ares_announcement("[MAIN_AI_SYSTEM] Comms Update", message)`
`log_ares_announcement("[title] Shipwide Update", message)`

**Review by Drulikar — CHANGES_REQUESTED** [review]
In ARES_interface_apollo.dm there is still:
`log_ares_security("Nerve Gas Release", "[last_login] released Nerve Gas from Vent '[sec_vent.vent_tag]'.")`

In ARES_interface_admin.dm there is still:
`log_ares_security("Nerve Gas Release", "[MAIN_AI_SYSTEM] released Nerve Gas from Vent '[sec_vent.vent_tag]'.")`

In apollo_pda.dm there is still:
`log_ares_security("Nerve Gas Release", "[last_login] released Nerve Gas from Vent '[sec_vent.vent_tag]'.")`

In biohazard_lockdown.dm there is still:
`var/ares_log = "[user.name] triggered Medical Research Biohazard Containment Lockdown."`
`log = "[key_name(user)] triggered research bio lockdown! (Using a custom announcement)."`
`ares_log = "[MAIN_AI_SYSTEM] triggered Medical Research Biohazard Containment Lockdown."`
`log = "[key_name(user)] lifted research bio lockdown!"`
`ares_log = "[user.name] lifted Medical Research Biohazard Containment Lockdown."`
`ares_log = "[MAIN_AI_SYSTEM] lifted Medical Research Biohazard Containment Lockdown."`
`log_ares_security("Containment Lockdown", ares_log)`

**Review by Drulikar — CHANGES_REQUESTED** [review]
Merge with master and you will find more procs you need to update as a result of https://github.com/cmss13-devs/cmss13/pull/6173

**Review by Drulikar — APPROVED** [review]
image

**Pull request merged**